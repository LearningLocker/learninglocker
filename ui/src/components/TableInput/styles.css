.table {
  margin-top: 0px; }

.table > thead > tr > th {
  padding-top: 10px;
  padding-bottom: 9px; }

.table > tbody > tr > td {
  padding: 2px 0px;
  vertical-align: middle; }
  
.table > tbody > tr > td input {
  border: none;
  box-shadow: none; }

.textRight {
  text-align: right; }

.textInput {
  padding: 1rem 0; }
