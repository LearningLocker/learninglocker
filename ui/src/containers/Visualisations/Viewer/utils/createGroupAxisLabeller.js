// @ts-check

/**
 * @param {{ [groupId: string ]: string }} groupDictionary
 * @returns {(groupId: string) => string}
 */
const createGroupAxisLabeller = (groupDictionary) => {
  return (groupId) => {
    const name = groupDictionary[groupId];
    return name !== undefined ? name : groupId;
  };
};

export default createGroupAxisLabeller;
