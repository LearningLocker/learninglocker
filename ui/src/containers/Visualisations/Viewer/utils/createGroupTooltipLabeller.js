// @ts-check

/**
 * @param {{ [groupId: string ]: string }} groupDictionary
 * @returns {(groupId: string) => string}
 */
const createGroupTooltipLabeller = (groupDictionary) => {
  return (groupId) => {
    const name = groupDictionary[groupId];
    return name !== undefined ? name : groupId;
  };
};

export default createGroupTooltipLabeller;
