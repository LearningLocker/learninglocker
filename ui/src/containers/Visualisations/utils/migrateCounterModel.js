export default (model) => {
  return {
    organisation: model.get('organisation'),
    owner: model.get('owner'),
    description: model.get('description'),
    isPublic: model.get('isPublic'),
    type: model.get('type'),
    config: {
      previewPeriod: model.get('previewPeriod'),
      filter: model.getIn(['filters', 0]),
      value: {
        key: model.getIn(['axesvalue', 'optionKey']),
        operator: model.get('axesoperator'),
        colour: 'red',
      },
    },
    createdAt: model.get('createdAt'),
    updatedAt: model.get('updatedAt'),
    _id: model.get('_id'),
  };
};
