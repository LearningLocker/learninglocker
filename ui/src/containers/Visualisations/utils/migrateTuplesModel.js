export default (model) => {
  return {
    organisation: model.get('organisation'),
    owner: model.get('owner'),
    description: model.get('description'),
    isPublic: model.get('isPublic'),
    type: model.get('type'),
    config: {
      previewPeriod: model.get('previewPeriod'),
      series: model.get('filters').map((series, seriesIndex) => {
        return {
          label: series.get('label', `Series ${seriesIndex}`),
          colour: series.get('color', 'orange'),
          filter: series.get('filter'),
        };
      }).toJS(),
      group: {
        label: model.getIn(['axesgroup', 'searchString']),
        key: model.getIn(['axesgroup', 'optionKey']),
      },
      firstValue: {
        label: model.getIn(['axesxValue', 'searchString']),
        key: model.getIn(['axesxValue', 'optionKey']),
        operator: model.get('axesxOperator'),
        filter: model.get('axesxQuery'),
      },
      secondValue: {
        label: model.getIn(['axesyValue', 'searchString']),
        key: model.getIn(['axesyValue', 'optionKey']),
        operator: model.get('axesyOperator'),
        filter: model.get('axesyQuery'),
      },
    },
    createdAt: model.get('createdAt'),
    updatedAt: model.get('updatedAt'),
    _id: model.get('_id'),
  };
};
