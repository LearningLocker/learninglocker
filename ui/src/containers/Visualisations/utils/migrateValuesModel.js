export default (model) => {
  return {
    organisation: model.get('organisation'),
    owner: model.get('owner'),
    description: model.get('description'),
    isPublic: model.get('isPublic'),
    type: model.get('type'),
    config: {
      previewPeriod: model.get('previewPeriod'),
      stacked: model.get('stacked'),
      series: model.get('filters').map((series, seriesIndex) => {
        return {
          label: series.get('label', `Series ${seriesIndex}`),
          colour: series.get('color', 'orange'),
          filter: series.get('filter'),
        };
      }).toJS(),
      group: {
        label: model.getIn(['axesgroup', 'searchString']),
        key: model.getIn(['axesgroup', 'optionKey']),
      },
      value: {
        label: model.getIn(['axesvalue', 'searchString']),
        key: model.getIn(['axesvalue', 'optionKey']),
        operator: model.get('axesoperator'),
      },
    },
    createdAt: model.get('createdAt'),
    updatedAt: model.get('updatedAt'),
    _id: model.get('_id'),
  };
};
