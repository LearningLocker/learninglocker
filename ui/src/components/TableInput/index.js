import React from 'react';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import { map } from 'lodash';
import DebounceInput from 'react-debounce-input';
import { Map } from 'immutable';
import {
  compose,
  withHandlers,
} from 'recompose';
import styles from './styles.css';

const defaultInputComponent = ({
  value,
  onChange
}) =>
  (
    <DebounceInput
      className="form-control"
      debounceTimeout={377}
      value={value}
      onChange={onChange} />
  );

const withHandlersEnhance = withHandlers(
  {
    onKeyChange: ({ values, onChange }) => key => (element) => {
      const newKey = element.target.value;

      const newValues = values.mapKeys(k =>
        ((k === key) ? newKey : k)
      );

      onChange(newValues);
    },
    onValueChange: props => key => (element) => {
      if (props.onValueChange) {
        return props.onValueChange(props)(key)(element);
      }

      const { values, onChange } = props;

      let newValue = element;
      if (element.target) {
        newValue = element.target.value;
      }

      onChange(
        values.set(key, newValue)
      );
    },
    remove: ({ values, onChange }) => key => () => {
      onChange(
        values.delete(key)
      );
    },
    add: ({ values, onChange }) => () => {
      onChange(
        values.set('New', '')
      );
    }
  }
);

const renderItems = ({
  staticValues,
  values,
  onKeyChange,
  onValueChange,
  inputComponent = defaultInputComponent,
  remove
}) => {
  const valuesJs = values.toJS();

  const staticValuesRendered = map(staticValues.toJS(), (value, key) =>
    (<tr>
      <td>
        <input className="form-control" type="text" value={key} disabled="true" />
      </td>
      <td>
        <input className="form-control" type="text" value={value} disabled="true" />
      </td>
    </tr>)
  );

  let trKey = 0;
  const valuesRendered = map(valuesJs, (value, key) => {
    if (key === '_id') {
      return null;
    }

    trKey += 1;

    return (<tr key={trKey}>
      <td>
        <DebounceInput
          className="form-control"
          debounceTimeout={377}
          value={key}
          onChange={onKeyChange(key)} />
      </td>
      <td>
        {inputComponent({
          key,
          onChange: onValueChange(key),
          value,
        })}
      </td>
      <td className={styles.textRight}>
        <a onClick={remove(key)} className="pull-right">
          <i className="icon ion-close" />
        </a>
      </td>
    </tr>);
  });
  return [staticValuesRendered, valuesRendered];
};

const render = ({
  keyName = 'Column',
  valueName = 'Source',
  staticValues = new Map(), // can't be changed
  values, // key value object
  onChange,
  remove,
  inputComponent,
  add,
  onKeyChange,
  onValueChange,
}) => {
  const items = renderItems({
    staticValues,
    values,
    onChange,
    onKeyChange,
    onValueChange,
    remove,
    inputComponent
  });

  return (<span>
    <table className={`table table-borderless ${styles.table}`}>
      <thead>
        <tr>
          <th>{keyName}</th>
          <th>{valueName}</th>
          <th />
        </tr>
      </thead>
      <tbody id="models">{items}</tbody>
    </table>
    <a className="btn btn-sm btn-inverse" style={{ width: '33px' }} onClick={add}>
      <i className="icon ion-plus" />
    </a>
  </span>);
};

export default compose(
  withStyles(styles),
  // withPropsOnChangeEnhance,
  withHandlersEnhance,
)(render);
