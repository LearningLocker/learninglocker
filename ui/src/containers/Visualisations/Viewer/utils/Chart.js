// @ts-check
import React from 'react';
import { LineChart, XAxis, YAxis, Bar, Legend, Tooltip } from 'recharts';
import { union } from 'lodash';
import { AutoSizer } from 'react-virtualized';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
// @ts-ignore
import styles from './styles.css';

/**
 * @typedef {Object} AutoSizerProps
 * @property {number} height
 * @property {number} width
 */

const render = withStyles(styles)(
  /**  @param {{ xAxisLabel: string, yAxisLabel: string, children: (props: AutoSizerProps) => React.Element }} props */
  (props) => {
    const { xAxisLabel, yAxisLabel, children } = props;
    return (
      <div className={styles.chart}>
        <div className={styles.xAxisLabel}>
          <span className={styles.xAxis}>
            {xAxisLabel || 'X Axis'}
          </span>
        </div>
        <div className={`${styles.barContainer}`}>
          <div className={styles.yAxisLabel}>
            <span className={styles.yAxis}>
              {yAxisLabel || 'Y Axis'}
            </span>
          </div>
          <div className={styles.chartWrapper}>
            <AutoSizer>
              {children}
            </AutoSizer>
          </div>
        </div>
      </div>
    );
  }
);

export default render;
