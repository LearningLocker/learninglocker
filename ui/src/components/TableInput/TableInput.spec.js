import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { withInsertCSS } from 'ui/utils/hocs';
import { fromJS } from 'immutable';
import TableInput from './index';

describe('TableInput', () => {
  const WrappedTableInput = withInsertCSS(TableInput);

  it('Should render table input', () => {
    const tableValue = fromJS({ test1: 'test2' });

    const tableInput = ReactTestRenderer.create(<WrappedTableInput
      values={tableValue}
      onChange={() => {}} />).toJSON();

    expect(tableInput).toMatchSnapshot();
  });
});
