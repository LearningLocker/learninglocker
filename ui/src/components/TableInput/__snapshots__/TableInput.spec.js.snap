// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableInput Should render table input 1`] = `
<span>
  <table
    className="table table-borderless table"
  >
    <thead>
      <tr>
        <th>
          Column
        </th>
        <th>
          Source
        </th>
        <th />
      </tr>
    </thead>
    <tbody
      id="models"
    >
      <tr>
        <td>
          <input
            className="form-control"
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            type="text"
            value="test1"
          />
        </td>
        <td>
          <input
            className="form-control"
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            type="text"
            value="test2"
          />
        </td>
        <td
          className="textRight"
        >
          <a
            className="pull-right"
            onClick={[Function]}
          >
            <i
              className="icon ion-close"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <a
    className="btn btn-sm btn-inverse"
    onClick={[Function]}
    style={
      Object {
        "width": "33px",
      }
    }
  >
    <i
      className="icon ion-plus"
    />
  </a>
</span>
`;
